src/core/init/duration_updater.ts

```diff
@@ -42,25 +62,29 @@ export default function DurationUpdater(
 ) : Observable<never> {
   return fromEvent(manifest, "manifestUpdate").pipe(
     startWith(null),
-    tap(() => {
-      const maximumPosition = manifest.getMaximumPosition();
+    switchMap(() => {
+      return onMediaSourceOpen$(mediaSource).pipe(
```

Comment:
Isn't that a little bit ugly, considering that it is almost always open?

It's maybe just the name but calling `onMediaSourceOpen` at each Manifest update does not seem logical to me

Reply:
Does getOpenedMediaSource$ or something like that seems to be a better solution for you ?

Reply:
I like it less, as there I would expect to get a MediaSource in response.

Maybe something more explicit like `lockUntilMediaSourceOpen$` / `waitForMediaSourceOpen$`?

Reply:
waitForMediaSourceOpen$ does not differ so much from onMediaSourceOpen$, no ?

Reply:
Functions prefixed by `on` make me think of the browser's event APIs (e.g. `onclick` and so on) which imply only future events.

Whereas `waitFor` is a little more ambiguous about what to do if the condition is already met. That's not perfect but it seems better to me

Reply:
Could also be `whenMediaSourceOpen`

Reply:
I just noticed that `whenMediaSourceOpen$` already exists in `src/compat/when_media_source_open.ts`.
I still thinks that's not the best name, but it's there.